http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/RequestAdapter.java

```diff
@@ -21,23 +21,79 @@
 import io.netty.handler.codec.http.HttpMethod;
 import io.netty.handler.codec.http.HttpRequest;
 import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.codec.http2.HttpConversionUtil.ExtensionHeaderNames;
+import java.util.Arrays;
+import java.util.List;
 import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.http.Protocol;
 import software.amazon.awssdk.http.SdkHttpMethod;
 import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.utils.StringUtils;
+import software.amazon.awssdk.utils.Validate;
+import software.amazon.awssdk.utils.http.SdkHttpUtils;
 
 @SdkInternalApi
 public final class RequestAdapter {
 
+    private static final String HOST = "Host";
+    private static final List<String> IGNORE_HEADERS = Arrays.asList(HOST);
+
+    private final Protocol protocol;
+
+    public RequestAdapter(Protocol protocol) {
+        this.protocol = Validate.paramNotNull(protocol, "protocol");
+    }
+
     public HttpRequest adapt(SdkHttpRequest sdkRequest) {
         HttpMethod method = toNettyHttpMethod(sdkRequest.method());
         HttpHeaders headers = new DefaultHttpHeaders();
-        String uri = sdkRequest.getUri().toString();
+        String uri = encodedPathAndQueryParams(sdkRequest);
         DefaultHttpRequest request = new DefaultHttpRequest(HttpVersion.HTTP_1_1, method, uri, headers);
```

Comment:
Should this also be updated to use the protocol instead of hard-coded `HTTP_1_1`?

Reply:
I thought about that but there is no `HttpVersion.HTTP2`. When we're on H2, the [conversion handler](https://github.com/aws/aws-sdk-java-v2/blob/240b78841d9169b6c5c7a5d1fcc129e141f8924e/http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/HttpToHttp2OutboundAdapter.java#L56) does the right thing.

Reply:
I see. Can we at least add a comment there?

Reply:
Sure added a comment.